packages/core/src/root.ts

```diff
@@ -360,6 +381,15 @@ export function createRemoteRoot<
           children: value.children.map((child) => serialize(child as any)),
         };
   }
+
+  function validateAllowedComponent(
+    componentType: AllowedComponents,
+    allowedComponents?: readonly AllowedComponents[],
```

Comment:
I think you could technically omit this argument since it's available in scope (unless I am reading incorrectly)

Reply:
I think this function is only called once. We might as well just inline the logic above:

```
if (allowedComponents && allowedComponents.indexOf(componentType) < 0) {
  throw new Error(`Unsupported component: ${type}`);
}
```